_1327/polls/views.py

```diff
@@ -53,6 +53,13 @@ def results(request, poll, url_title):
 				and poll.end_date > datetime.date.today():
 		return vote(request, poll, url_title)
 
+	if not poll.show_results_immediately and poll.end_date > datetime.date.today():
```

Comment:
please use consistently `poll.end_date >= datetime.date.today()` to check whether the voting period is still in progress. users should be able to vote until and including the `end_date`